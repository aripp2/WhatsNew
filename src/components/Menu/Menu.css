aside {
  color: blue;
  width: 25%;
  display: flex;
  flex-direction: column;
}

.menu-btn {
  color: #141527;
  font-size: 1.5em;
  height: 40px;
  width: 60%;
  border-radius: 5px;
  margin: 20px auto;
}