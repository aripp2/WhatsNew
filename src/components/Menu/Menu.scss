@import './src/variables.scss';

nav {
  width: 20%;
  color: $dark-color;
  display: flex;
  flex-direction: column;
}

h1 {
  font-size: 2.5em;
  margin: 20px auto;
}

li {
  list-style: none;
}

.menu-btn {
  height: 60px;
  width: 90%;
  color: $light-color;
  font-size: 1.5em;
  text-align: left;
  margin: 20px;
  background-color: $dark-color;
  border: none;
  border-radius: 5px;
  display: flex;
  align-items: center;
  justify-content: left;

  svg {
    height: 1.5em;
    width: 1.5em;
    fill: $light-color;
    margin-right: 15px;
    background-color: $dark-color;
  }

  &:hover {
    background-color: $highlight-color;
    
    svg {
      background-color: $highlight-color;
    }
  }
}

