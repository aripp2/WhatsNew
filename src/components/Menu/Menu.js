import React from 'react';
import './Menu.scss';
import { FaMapMarkerAlt } from 'react-icons/fa';
import { FaLaptop } from 'react-icons/fa';
import { FaFilm } from 'react-icons/fa';
import { FaFlask } from 'react-icons/fa';
import { FaHeartbeat } from 'react-icons/fa';

const Menu = ({ changeNewsType }) => {
  return (
    <nav>
      <h1>What's New?</h1>
      <ul>
        <li>
          <button 
            className="menu-btn"
            onClick={() => changeNewsType('local')}
            ><FaMapMarkerAlt />Local News</button>
        </li>
        <li>
          <button 
            className="menu-btn"
            onClick={() => changeNewsType('technology')}
            ><FaLaptop />Technology</button>
        </li>
        <li>
          <button 
            className="menu-btn"
            onClick={() => changeNewsType('entertainment')}
            ><FaFilm />Entertainment</button>
        </li>
        <li>
          <button 
            className="menu-btn"
            onClick={() => changeNewsType('science')}
            ><FaFlask />Science</button>
        </li>
        <li>
          <button 
            className="menu-btn"
            onClick={() => changeNewsType('health')}
            ><FaHeartbeat />Health</button>
        </li>
      </ul>
    </nav>
  )
}

export default Menu;