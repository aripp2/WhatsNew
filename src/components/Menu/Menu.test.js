import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {

  let wrapper;
  const mockChangeNewsType=jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Menu
      changeNewsType={mockChangeNewsType}
     />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should call the changeNewsType prop with the newsType when clicked', () => {

    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(4).simulate('click');

    expect(mockChangeNewsType).toHaveBeenCalledWith('local');
    expect(mockChangeNewsType).toHaveBeenCalledWith('health');

  });

});

