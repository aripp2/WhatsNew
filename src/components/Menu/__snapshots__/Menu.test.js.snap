// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<nav>
  <h1>
    What's New?
  </h1>
  <ul>
    <li>
      <button
        className="menu-btn"
        onClick={[Function]}
      >
        <FaMapMarkerAlt />
        Local News
      </button>
    </li>
    <li>
      <button
        className="menu-btn"
        onClick={[Function]}
      >
        <FaLaptop />
        Technology
      </button>
    </li>
    <li>
      <button
        className="menu-btn"
        onClick={[Function]}
      >
        <FaFilm />
        Entertainment
      </button>
    </li>
    <li>
      <button
        className="menu-btn"
        onClick={[Function]}
      >
        <FaFlask />
        Science
      </button>
    </li>
    <li>
      <button
        className="menu-btn"
        onClick={[Function]}
      >
        <FaHeartbeat />
        Health
      </button>
    </li>
  </ul>
</nav>
`;
